logic/peer.js

```diff
@@ -36,25 +51,35 @@ Peer.prototype.accept = function (peer) {
 		this.ip = peer.ip;
 	}
 
-	this.port = this.parseInt(peer.port, 0);
+	this.port = self.parseInt(peer.port, 0);
 
-	if (this.ip) {
-		this.string = (this.ip + ':' + this.port || 'unknown');
-	} else {
-		this.string = 'unknown';
+	if (this.ip && this.port) {
+		this.string = this.ip + ':' + this.port;
 	}
 
-	if (peer.state != null) {
+	if (/^[0-2]{1}$/.test(peer.state)) {
 		this.state = peer.state;
 	} else {
 		this.state = 1;
 	}
 
-	if (peer.dappid != null) {
-		this.dappid = peer.dappid;
+	if (peer.dappid) {
+		if (Array.isArray(peer.dappid)) {
+			this.dappid = peer.dappid;
+		} else {
+			this.dappid = [];
+			this.dappid.push(peer.dappid);
+		}
+	}
+
+	if (peer.height) {
+		this.height = self.parseInt(peer.height, 1);
+	}
+
+	if (peer.clock) {
+		this.clock = peer.clock;
 	}
 
```

Comment:
Nothing, those values can be set only via `applyHeaders`, but I will re-consider that.